crates/runtime/src/flight/flightsql/prepared_statement_query.rs

```diff
@@ -102,19 +124,153 @@ pub(crate) async fn do_get(
 
     let datafusion = Arc::clone(&flight_svc.datafusion);
     tracing::trace!("do_get: {query:?}");
-    match std::str::from_utf8(&query.prepared_statement_handle) {
-        Ok(sql) => {
-            let (output, from_cache) =
-                Box::pin(Service::sql_to_flight_stream(datafusion, sql)).await?;
-            let timed_output = TimedStream::new(output, move || start);
-
-            let mut response =
-                Response::new(Box::pin(timed_output) as <Service as FlightService>::DoGetStream);
-            attach_cache_metadata(&mut response, from_cache);
-            Ok(response)
+
+    let handle =
+        String::from_utf8(query.prepared_statement_handle.to_vec()).map_err(error_to_status)?;
+
+    let (sql, parameters) = match &flight_svc.prepared_statements.read().await.get(&handle) {
+        Some(PreparedStatement { query, parameters }) => (query.clone(), parameters.clone()),
+        None => (handle, None),
+    };
+
+    let (output, from_cache) =
+        Box::pin(Service::sql_to_flight_stream(datafusion, &sql, parameters)).await?;
+    let timed_output = TimedStream::new(output, move || start);
+
+    let mut response =
+        Response::new(Box::pin(timed_output) as <Service as FlightService>::DoGetStream);
+    attach_cache_metadata(&mut response, from_cache);
+    Ok(response)
+}
+
+pub(crate) async fn do_put_query(
```

Comment:
nit: Add a comment that explains this is what "binds" the parameters to the prepared statement.